bunsen-spark/src/main/java/com/cerner/bunsen/spark/codes/base/AbstractValueSets.java

```diff
@@ -153,6 +153,11 @@ protected AbstractValueSets(SparkSession spark,
     return spark.createDataset(members.rdd(), URL_AND_VERSION_ENCODER)
         .collectAsList()
         .stream()
+        .peek(p -> {
+          if (p.getVersion() == null) {
+            throw new RuntimeException("Version is null for valueset with url " + p.getUrl());
```

Comment:
can `getUrl()` be `null`, in that case, will be any other information which we should log part of the message.

Reply:
We can look for `name` or `title` but none of them or mandatory from FHIR specifications so cannot be guaranteed to be present. I think we can stick to URL as this is the unique key element that's used in retrieving the valueset info programmatically, in most of the use cases.

Reply:
Thanks @amarvakul . But is `url` is a mandatory field, if it is `null`, then the message is basically of no use :-(

Reply:
Right @karthisuku, `url` is also not a mandatory field and when it is also not valued, the error message may not be completely helpful, but in the context where the error is currently being thrown (in `BroadcastableValuesets`), would at least give a lead/pointer that version and url are both empty.

Reply:
Just a thought, Can we log the entire object, which would help us to identify which object is failing for triage purpose.

Reply:
@karthisuku What object are you referring to? If its UrlAndVersion then it has only 2 fields { url, version} which are logged in the error message.

Reply:
Discussed offline, At this point during `peek` it does not have any more information otherthan `url` `version`.  In future if additional details are available we will enhance the logger as well.